plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestTrinoHiveCatalog.java

```diff
@@ -0,0 +1,98 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.plugin.base.CatalogName;
+import io.trino.plugin.hive.HdfsConfig;
+import io.trino.plugin.hive.HdfsConfiguration;
+import io.trino.plugin.hive.HdfsConfigurationInitializer;
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.HiveHdfsConfiguration;
+import io.trino.plugin.hive.authentication.NoHdfsAuthentication;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.iceberg.catalog.TrinoCatalog;
+import io.trino.plugin.iceberg.catalog.file.FileMetastoreTableOperationsProvider;
+import io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog;
+import io.trino.spi.type.TestingTypeManager;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestTrinoHiveCatalog
+        extends BaseTestTrinoCatalog
+{
+    private final HiveMetastore metastore;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final java.nio.file.Path tempDir;
+    private final File metastoreDir;
+
+    public TestTrinoHiveCatalog()
+            throws IOException
+    {
+        HdfsConfig config = new HdfsConfig();
+        HdfsConfiguration configuration = new HiveHdfsConfiguration(new HdfsConfigurationInitializer(config), ImmutableSet.of());
+        hdfsEnvironment = new HdfsEnvironment(configuration, config, new NoHdfsAuthentication());
+
+        tempDir = Files.createTempDirectory("test_iceberg_v2");
+        metastoreDir = tempDir.resolve("iceberg_data").toFile();
+        metastore = createTestingFileHiveMetastore(metastoreDir);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(tempDir, ALLOW_INSECURE);
+    }
+
+    @Override
+    protected TrinoCatalog createTrinoCatalog(boolean useUniqueTableLocations)
+    {
+        return new TrinoHiveCatalog(
+                new CatalogName("catalog"),
+                metastore,
+                hdfsEnvironment,
+                new TestingTypeManager(),
+                new FileMetastoreTableOperationsProvider(new HdfsFileIoProvider(hdfsEnvironment)),
+                "trino-version",
+                useUniqueTableLocations,
+                false,
+                false);
+    }
+
+    @Override
+    @Test
+    public void testCreateNamespaceWithLocation()
+    {
+        assertThatThrownBy(super::testCreateNamespaceWithLocation)
+                .hasMessageContaining("Database cannot be created with a location set");
```

Comment:
yikes, why?
it sounds like an important feature

Reply:
It is supported on HMS and Glue but not FileHiveMetastore